lava-provider/README.md

```diff
@@ -0,0 +1,53 @@
+# Deploy Lava Provider
+
+The launch is described using CloudFlare!
+
+## Step 1 (Deploy)
+
+> "How to use?" CloudMos guide: https://github.com/DecloudNodesLab/Guides/blob/main/English/Cloudmos.md
+
+Deploy https://gitopia.com/DecloudNodesLab/cosmos-universe/tree/master/projects/Lava/lava_provider_deploy.yml .
+
+Set  variables `SSH_PASS`, `MNEMONIC_BASE64`, `CONFIG_LINK`, `LAVAD_NODE` and `LAVAD_GEOLOCATION`.
+
+![image](https://github.com/DecloudNodesLab/Projects/assets/23629420/2c56f21b-7ef3-41aa-a665-2460ca261865)
+
+Select provider.
+
+![image](https://github.com/DecloudNodesLab/Projects/assets/23629420/cb1f53a0-286c-465f-9dd9-4a3948579570)
+
+## Step 2 (CloudFlare)
+
+Go to https://dash.cloudflare.com/ , select your domain name.
+
+In left menu:
+
+- Add "DNS" records TYPE - A, NAME - \<SUBDOMAIN\>, CONTENT - <YOUR_LEASE_IP_FROM_DEPLOY>
+
+![image](https://github.com/DecloudNodesLab/Projects/assets/23629420/774d791c-afdf-4724-973f-0c16504154cf)
+
+- Check "SSL/TLS-Overwiew", need select Full mode (not Full(strict)).
+
+![image](https://github.com/DecloudNodesLab/Projects/assets/23629420/0f57098f-bc61-4093-8cd5-3136d072e073)
+
+- In "Network" enable "gRPS" connection.
+
+![image](https://github.com/DecloudNodesLab/Projects/assets/23629420/b9fc19e0-0015-4484-838d-e764672e6c95)
+
+## Step 3 (run rpcprovider)
+
+After that, it will be convenient to connect to the node via **ssh**.
+
+Run test command:
+
+`lavad test rpcprovider --from $ADDRESS --keyring-backend test --node $LAVAD_NODE --endpoints "<YOUR_SUBDOMAIN.DOMAIN>:443,<CHAIN>"`
+
+Example correct result:
+
+![image](https://github.com/DecloudNodesLab/Projects/assets/23629420/c67de1a0-0099-4a07-9db2-6446290302d0)
+
+Run stake command:
+
+```
+lavad tx pairing stake-provider "<CHAIN>" "<AMOUNT_STAKE>ulava" "lava.declab.pro:443,2" 2 --chain-id $CHAIN_ID --from $ADDRESS --provider-moniker <YOUR_MONIKER> --keyring-backend "test" --gas="auto" --gas-adjustment "1.5" --fees 50000ulava --node $LAVAD_NODE -y
```

Comment:
I think you do not need to specify the `--fees`  when you specify `--gas="auto" --gas-adjustment "1.5"` (i.e. they cancel out each other);

Also `--gas="auto" --gas-adjustment "1.5"` won't take any effect unless you set `--gas-prices <value>`.

See how fees are getting calculated as well as how gas=auto gets to calculate them from my previous reply in the chat we had the other day:

----

please don't mix the `--minimum-gas-prices` (validators are asking for including the txs into the block) with the `--gas-prices` (clients are paying to the network [to the validators] so their txs aren't running `out of gas`)

* `--minimum-gas-prices`: **validators** set the _minimum_ gas prices
```
$ lavad start --help |grep price
      --minimum-gas-prices string                       Minimum gas prices to accept for transactions; Any fee in a tx must meet this minimum (e.g. 0.01photino;0.0001stake)
```

* `--gas-prices`: **clients** set the gas prices parameter to determine the transaction fee (the fee are about to pay for the tx to be accepted by the network (validators) so to not run `out of gas`):

```
$ lavad tx bank send --help |grep -- '  --gas-prices'
      --gas-prices string        Gas prices in decimal format to determine the transaction fee (e.g. 0.1uatom)
```

FWIW, the clients they can also control the gas calculation when the network is busy either by specifying  `--gas auto` along with the `--gas-adjustment` (low chance of getting a failed tx due to `out of gas`) or directly setting the `--gas` to a previously known value (high chance of the tx running `out of gas`).

```$ lavad tx bank send --help |grep -- '  --gas'
      --gas string               gas limit to set per-transaction; set to "auto" to calculate sufficient gas automatically. Note: "auto" option doesn't always report accurate results. Set a valid coin value to adjust the result. Can be used instead of "fees". (default 200000)
      --gas-adjustment float     adjustment factor to be multiplied against the estimate returned by the tx simulation; if the gas limit is set manually this flag is ignored  (default 1)
```

Additionally, here is how fees are calculated:

`fees = (gas * gas-prices)` as per Cosmos SDK docs.
`(200000 * 0.0025ulava) = 500ulava` floor fee for most of RPC nodes.  (`200000` is the default gas value which varies depending on the network load, e.g. when someone is sending lots of txs the gas value parameter automatically increases; BlockGasMeter watches and controls that https://docs.cosmos.network/main/basics/gas-fees#block-gas-meter)

With the `--gas auto --gas-adjustment 1.2` (which I always recommend using):
`fees = (gas * gas-prices) * gas-adjustment`
`fees = (200000 * 0.0025ulava) * 1.2 = 600ulava`

Note that RPC node running without minimum-gas-prices set will accept 0ulava fee transactions which is not desirable as per Cosmos SDK docs.

Source: https://docs.cosmos.network/main/basics/gas-fees